src/main/java/com/cloudbees/hudson/plugins/folder/properties/FolderCredentialsProvider.java

```diff
@@ -505,7 +505,7 @@ private synchronized boolean addCredentials(@NonNull Domain domain, @NonNull Cre
                 if (list == null || list.isEmpty()) {
                     return Collections.emptyList();
                 }
-                return Collections.unmodifiableList(new ArrayList<Credentials>(list));
+                return Collections.unmodifiableList(list);
```

Comment:
on the surface this seems wrong as the list returned my `getDomainCredentialsMap().get(domain);` may not be immutable, thus the unmodifiable collection could actually be modified after it was returned.

```suggestion
                return Collections.unmodifiableList(new ArrayList<>(list));
```